tools/schemacode/schema.py

```diff
@@ -47,8 +79,12 @@ def load_schema(schema_path):
     """
     schema_path = Path(schema_path)
     if schema_path.is_file() and (schema_path.suffix == ".yaml"):
+        base_path = os.path.dirname(schema_path.absolute())
         with open(schema_path) as f:
-            return yaml.load(f, Loader=yaml.SafeLoader)
+            dict_ = yaml.load(f, Loader=yaml.SafeLoader)
+            if "$ref" in str(dict_):
+                dict_ = search_structure(dict_, base_path)
+            return dict_
```

Comment:
Seems an unnecessary decomposition. And if we change to `derefence_yaml(yaml.load(...))` I think it will be very clear.

```suggestion
            return search_structure(yaml.load(f, Loader=yaml.SafeLoader))
```

Reply:
My goal here is to speed things up. Calling the recursive `search_structure` function (now `dereference_yaml`) on a YAML file without `$ref` in it seems like it would slow things down with no benefit.

Reply:
Have you measured it? `str()` is already recursing. My suspicion is that it's such a trivial amount of time when there aren't `$ref`s to deref that there's essentially no penalty.

Reply:
You're right. Your way is definitely faster.